src/bin/commands/rustc.rs

```diff
@@ -62,7 +62,12 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
         }
     };
     let mut compile_opts = args.compile_options_for_single_package(config, mode)?;
-    compile_opts.target_rustc_args = Some(values(args, "args"));
+    let target_args = values(args, "args");
+    compile_opts.target_rustc_args = if target_args.is_empty() {
```

Comment:
Was this change needed for the correctness of the patch? I think it changes the behavior of `cargo rustc` by default (erroring today whereas after this patch it succeeds and builds multiple crates)

Reply:
Yes, this is needed for e.g. `cargo rustc` with no args to work in the first place.

Assuming "package" and "crate" are the same thing, the behaviour you're describing is tested for by `fail_with_multiple_packages` which still passes. There is also a `fails_when_trying_to_build_main_and_lib_with_args` test for failing when multiple targets are selected, and it is still passing as well.

Reply:
> Yes, this is needed for e.g. `cargo rustc` with no args to work in the first place.

Sorry, this is not correct. The hunk here is needed for `cargo rustc --all-targets` with no further args to work, which is documented as part of its CLI. Furthermore it's also needed when passing cargo args like `-v`, as opposed to binary-target args that go after `--` which is what `fails_when_trying_to_build_main_and_lib_with_args` tests for and this is still passing.

Reply:
Hm I think the documentation is wrong then? The `cargo rustc` command was only ever intended to work with only one target

Reply:
Ok but my commit here doesn't affect this (i.e. it was broken before and it's broken afterwards), it only fixes the `--` related behaviour that was broken by #5152. If you run `cargo rustc` using cargo 0.25.0 (the current stable) then it will still work with a multi-target crate, e.g.

~~~~
$ cat Cargo.toml 
[package]
name = "foo"
version = "0.0.1"
authors = []

$ cat src/lib.rs 
pub fn myfunc() {}

$ cat src/main.rs 
fn main () {}

$ cargo --version
cargo 0.25.0

$ cargo rustc --verbose
   Compiling foo v0.0.1 (file://$PWD)
     Running `rustc --crate-name foo src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=0fa9827c683f047b -C extra-filename=-0fa9827c683f047b --out-dir $PWD/target/debug/deps -L dependency=$PWD/target/debug/deps`
     Running `rustc --crate-name foo src/main.rs --crate-type bin --emit=dep-info,link -C debuginfo=2 -C metadata=a9125b8469059127 -C extra-filename=-a9125b8469059127 --out-dir $PWD/target/debug/deps -L dependency=$PWD/target/debug/deps --extern foo=$PWD/target/debug/deps/libfoo-0fa9827c683f047b.rlib`
    Finished dev [unoptimized + debuginfo] target(s) in 0.35 secs
~~~~

Reply:
Ah I see... odd! Well in any case it looks like nothing's changing so seems good to land!